### ISSUE TITLE
Question on ACME `credentialsFile` Structure

### ISSUE THREAD

**Issue by enricozb** [Xe/nixos-configs#4]
## TLDR
Do you mind sharing the _structure_ of your the credentials file referenced here: https://github.com/Xe/nixos-configs/blob/f25188ed1e61dcd7e29fd10547aff69a6b0ab4ae/hosts/lufta/acme.nix#L6 Of course I'm not looking to grab your keys, I just want to know what the nix expression in there looks like.

# Full Question
First of all, thank you for publishing your nix configs so we can reference them.

I'm trying to set up ACME similar to you, I have cloudflare DNS, and want to do the DNS challenge for the certs (for wildcards), _and_ I want to use Cloudflare's API tokens, as opposed to their keys.

I can't get nix to accept what I put in `security.acme.certs.credentialsFile`. I originally tried
```
CLOUDFLARE_DNS_API_TOKEN=...
```
like a common `.env` file. But it looks like the ACME module tries to evaluate that file as a nix expression, so I tried
```
{
  CLOUDFLARE_DNS_API_TOKEN = ...;
}
```
but then nix complains that option doesn't exist under `credentialsFile`.

Do you mind sharing the structure of your the credentials file referenced here: https://github.com/Xe/nixos-configs/blob/f25188ed1e61dcd7e29fd10547aff69a6b0ab4ae/hosts/lufta/acme.nix#L6

Thanks in advance.

**Comment by Xe** [comment]
It looks like this:

```shell
CLOUDFLARE_EMAIL=foo@bar.com
CLOUDFLARE_API_KEY=hunter2hunter2hunter2
```

**Comment by enricozb** [comment]
Hmm, ok it must be something with my config then. For some reason my env file is being evaluated as if it were a nix expression. Thanks for confirming that!

**Comment by Xe** [comment]
When you find out what the problem is, please let me know or write something up on it; if only so people in the future have something to go on when debugging this issue.

**Comment by enricozb** [comment]
I ended up starting from scratch with the self-hosted config using `bind` from the docs, and slowly moved to my final config using the cloudflare creds. I was trying to track down exactly what was happening, but my knowledge of nix isn't good enough to really trace the execution of the language. `--show-trace` wasn't that helpful, but I imagine this might be partly because the nix language is lazy. Either way, I'll try to reproduce it again and get an MVP broken config.

In case it helps anyone else out, here's the working config I ended up with:
```
  security.acme = {
    acceptTerms = true;
    defaults.email = "enricozb+acme-le@gmail.com";

    certs."ezb.dev" = {
      group = "nginx";
      dnsProvider = "cloudflare";
      credentialsFile = "/var/lib/secrets/certs.secret";
      extraDomainNames = [ "*.ezb.dev" ];
      extraLegoFlags = [ "--dns.resolvers=8.8.8.8:53" ];
    };
  };

  services.nginx.enable = true;

  services.nginx.virtualHosts."serve.ezb.dev" = {
    forceSSL = true;
    useACMEHost = "ezb.dev";
    locations."/" = { proxyPass = "http://127.0.0.1:25704"; };
  };
```

Also I thought [you were using](https://discourse.nixos.org/t/weird-issue-with-acme-and-cloudflare-dns/13123) the cloudflare API tokens (asking because I see `CLOUDFLARE_API_KEY` in the .env you sent)?
